### PR TITLE
typo fix in HaxeManual/02-types.tex

### PR DESCRIPTION
**Pull request by iamrajiv**

### CODE REVIEW

File: HaxeManual/02-types.tex [hunk 1]
```diff
@@ -437,7 +437,7 @@ \subsection{Using enums}
 \section{Anonymous Structure}
 \label{types-anonymous-structure}
 
-Anonymous structures can be used to group data without explicitly creating a type. The following example creates a structure with two fields \expr{x} and \expr{name}, and initializes their values to \expr{12} and \expr{"foo"} respectively:
+Anonymous structures can be used to group data without explicitly creating a type. The following example creates a structure with two fields \expr{x} and \expr{name} and initializes their values to \expr{12} and \expr{"foo"} respectively:
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: I think the comma should be kept here, to not have an ambiguous `x and y and z`

File: HaxeManual/02-types.tex [hunk 2]
```diff
@@ -597,7 +597,7 @@ \subsection{Extensions}
 
 \since{4.0.0}
 
-Haxe 4 introduces an alternative notation for extension, separating each extended structure with a \expr{\&} symbol.
+Haxe 4 introduces an alternative notation for an extension, separating each extended structure with a \expr{\&} symbol.
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: Unsure about this. If the original sounds too wrong I would go for `alternative notation for extensions`

File: HaxeManual/02-types.tex [hunk 3]
```diff
@@ -678,12 +678,12 @@ \subsection{Default values}
 \section{Dynamic}
 \label{types-dynamic}
 
-While Haxe has a static type system, this type system can, in effect, be turned off by using the \type{Dynamic} type. A \emph{dynamic value} can be assigned to anything; and anything can be assigned to it. This has several drawbacks:
+While Haxe has a static type system, this type of system can, in effect, be turned off by using the \type{Dynamic} type. A \emph{dynamic value} can be assigned to anything and anything can be assigned to it. This has several drawbacks:
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: Dropping the semicolon is ok I suppose, but it is a `type system`. It is not a general property of "this type of systems" to be able to turn them off with a `Dynamic` type.

File: HaxeManual/02-types.tex [hunk 4]
```diff
@@ -678,12 +678,12 @@ \subsection{Default values}
...
 
 \begin{itemize}
 	\item The compiler can no longer type-check assignments, function calls and other constructs where specific types are expected.
 	\item Certain optimizations, in particular when compiling to static targets, can no longer be employed.
-	\item Some common errors, e.g. a typo in a field access, can not be caught at compile-time and likely cause an error at runtime.
+	\item Some common errors, e.g. a typo infield access, cannot be caught at compile-time and likely cause an error at runtime.
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: `infield` -> `in field` ?

File: HaxeManual/02-types.tex [hunk 5]
```diff
@@ -734,15 +734,15 @@ \subsection{Any type}
 \label{types-dynamic-any}
 
 \type{Any} is a type that is compatible with any other in both ways. 
-It serves one purpose - to hold values of any type, but to actually use that values, an explicit casting is required. That way the code doesn't suddently become dynamically typed and we keep all the static typing goodness, like advanced type system features and optimizations.
+It serves one purpose - to hold values of any type but to actually use that values, explicit casting is required. That way the code doesn't suddenly become dynamically typed and we keep all the static typing goodness, like advanced type system features and optimizations.
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: The comma after `type` should remain – those are two independent clauses. `that values` -> `those values`.

File: HaxeManual/02-types.tex [hunk 6]
```diff
@@ -779,14 +779,14 @@ \section{Abstract}
...
 
 
 
-\trivia{Basic Types and abstracts}{Before the advent of abstract types, all basic types were implemented as extern classes or enums. While this nicely took care of some aspects such as \type{Int} being a ``child class'' of \type{Float}, it caused issues elsewhere. For instance, with \type{Float} being an extern class, it would unify with the empty structure \expr{\{\}}, making it impossible to constrain a type to accepting only real objects.}
+\trivia{Basic Types and abstracts}{Before the advent of abstract types, all basic types were implemented as extern classes or enums. While this nicely taken care of some aspects such as \type{Int} being a ``child class'' of \type{Float}, it caused issues elsewhere. For instance, with \type{Float} being an extern class, it would unify with the empty structure \expr{\{\}}, making it impossible to constrain a type to accept only real objects.}
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: `took` is correct.

File: HaxeManual/02-types.tex [hunk 7]
```diff
@@ -808,7 +808,7 @@ \subsection{Implicit Casts}
 Class field casts have the same semantics, but are defined completely differently:
 
 \haxe{assets/ImplicitCastField.hx}
-By adding \expr{@:from} to a static function, that function qualifies as implicit cast function from its argument type to the abstract. These functions must return a value of the abstract type. They must also be declared \expr{static}.
+By adding \expr{@:from} to a static function, that function qualifies as implicit cast function from its argument type to the abstract. These functions must return the value of the abstract type. They must also be declared \expr{static}.
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: No, they return *a* value that is of the abstract type, i.e. a new instance of that abstract.

File: HaxeManual/02-types.tex [hunk 8]
```diff
@@ -935,7 +935,7 @@ \subsection{Array Access}
 
 The array access \expr{a[0]} is resolved to the \expr{getInt1} field, leading to lower case \expr{f} being returned. The result might be different in Haxe versions before 3.2.
 
-Fields which are defined earlier take priority even if they require an \tref{implicit cast}{types-abstract-implicit-casts}.
+Fields which are defined earlier take priority even if they require a \tref{implicit cast}{types-abstract-implicit-casts}.
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: No.

File: HaxeManual/02-types.tex [hunk 9]
```diff
@@ -974,10 +974,10 @@ \subsection{Enum abstracts}
 
 \since{4.0.0}
 
-Since Haxe 4, it is possible to declare enum abstracts without using the \expr{:enum} metadata, instead using the more natural syntax \expr{enum abstract}. Additionally, if the underlying type is \expr{String} or \expr{Int}, the values for the enum cases can be omitted and are deduced by the compiler:
+Since Haxe 4, it is possible to declare enum abstracts without using the \expr{:enum} metadata, instead of using the more natural syntax \expr{enum abstract}. Additionally, if the underlying type is \expr{String} or \expr{Int}, the values for the enum cases can be omitted and are deduced by the compiler:
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: No, this changes the meaning of the sentence. If the original sounds wrong, maybe `... metadata, using the more natural syntax \expr{enum abstract} instead.`

File: HaxeManual/02-types.tex [hunk 10]
```diff
@@ -1014,7 +1014,7 @@ \subsection{Core-type abstracts}
 
 The Haxe Standard Library defines a set of basic types as core-type abstracts. They are identified by the \expr{:coreType} metadata and the lack of an underlying type declaration. These abstracts can still be understood to represent a different type. Still, that type is native to the Haxe target. 
 
-Introducing custom core-type abstracts is rarely necessary in user code as it requires the Haxe target to be able to make sense of it. However, there could be interesting use-cases for authors of macros and new Haxe targets.
+Introducing custom core-type abstracts is rarely necessary for user code as it requires the Haxe target to be able to make sense of it. However, there could be interesting use-cases for authors of macros and new Haxe targets.
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: I think I like the original better here.

File: HaxeManual/02-types.tex (hunk 2 above)
Comment (Gama11) [reply HaxeManual/02-types.tex]: I think the original sounds just fine.

Comment (Simn) [reply HaxeManual/02-types.tex]: "for extensions" sounds better

**Comment by iamrajiv** [comment]
thank you @Aurel300 for correcting my mistakes and for your suggestions

File: HaxeManual/02-types.tex [hunk 11]
```diff
@@ -808,7 +808,7 @@ \subsection{Implicit Casts}
 Class field casts have the same semantics, but are defined completely differently:
 
 \haxe{assets/ImplicitCastField.hx}
-By adding \expr{@:from} to a static function, that function qualifies as implicit cast function from its argument type to the abstract. These functions must return a value of the abstract type. They must also be declared \expr{static}.
+By adding \expr{@:from} to a static function, that function qualifies as implicit cast function from its argument type to the abstract. These functions must return the value of a abstract type. They must also be declared \expr{static}.
```
Comment (Aurel300) [inline HaxeManual/02-types.tex]: No, `a value of the abstract type` :)

**Pull request merged**